src/components/Widgets/Markdown/serializers/remarkSlate.js

```diff
@@ -0,0 +1,293 @@
+import { get, isEmpty, isArray } from 'lodash';
+import u from 'unist-builder';
+
+/**
+ * Map of MDAST node types to Slate node types.
+ */
+const typeMap = {
+  root: 'root',
+  paragraph: 'paragraph',
+  blockquote: 'quote',
+  code: 'code',
+  listItem: 'list-item',
+  table: 'table',
+  tableRow: 'table-row',
+  tableCell: 'table-cell',
+  thematicBreak: 'thematic-break',
+  link: 'link',
+  image: 'image',
+  shortcode: 'shortcode',
+};
+
+
+/**
+ * Map of MDAST node types to Slate mark types.
+ */
+const markMap = {
+  strong: 'bold',
+  emphasis: 'italic',
+  delete: 'strikethrough',
+  inlineCode: 'code',
+};
+
+
+/**
+ * Create a Slate Inline node.
+ */
+function createBlock(type, nodes, props = {}) {
+  if (!isArray(nodes)) {
+    props = nodes;
+    nodes = undefined;
+  }
+
+  return { kind: 'block', type, nodes, ...props };
+}
+
+
+/**
+ * Create a Slate Block node.
+ */
+function createInline(type, nodes, props = {}) {
+  return { kind: 'inline', type, nodes, ...props };
+}
+
+
+/**
+ * Create a Slate Raw text node.
+ */
+function createText(value, data) {
+  const node = { kind: 'text', data };
+  if (isArray(value)) {
+    return { ...node, ranges: value };
+  }
+  return {...node, text: value };
+}
+
+function convertMarkNode(node, parentMarks = []) {
+
+  /**
+   * Add the current node's mark type to the marks collected from parent
+   * mark nodes, if any.
+   */
+  const markType = markMap[node.type];
+  const marks = markType ? [...parentMarks, { type: markMap[node.type] }] : parentMarks;
+
+  /**
+   * Set an array to collect sections of text.
+   */
+  const ranges = [];
+
+  node.children && node.children.forEach(childNode => {
```

Comment:
You should be able to replace this loop with `Array.filter` and [`_.flatMap`](https://lodash.com/docs/4.17.4#flatMap) and remove the mutation. If the array mutation is here for performance reasons, further improvement could be made by using a `for` loop rather than `.forEach`, which would remove function call overhead. OTOH this is quite readable as it stands if you'd prefer to keep the current implementation.